### PR TITLE
Class and slot names normalized according to LinkML linter

### PR DESCRIPTION
**Pull request by turbomam**
For issue #456

This is a long overdue normalization of the nmdc-schema class and slot names. Slots now use use snake_case and classes use UpperCamelCase aka PascalCase. Brought to you by @pkalita-lbl's LinkML linter.

This will broadly improve the  usability of the schema, way beyond making things more internally consistent and visually differentiate-able. One example: efficient searching of the documentation pages currently requires users to know about silent repairs the doc generator may make to slot and class names. Is it "gold sample identifiers"? "GoldSampleIdentifiers"? or "gold_sample_identifiers"? Good luck finding your term of interest in the documentation pages!

The internal tests are consistent with the revised schema, but nothing external will be compatible yet. That includes MongoDB, the Data Portal, and any supporting code. Since those are already using a schema that's behind by three major releases, I don't feel like these new changes are all that radical. I plan on doing a conversion of the `biosample_set` collection within MongoDB soon as an illustration.

There is currently no documentation of the changes other than [this PR's Files Changed tab](https://github.com/microbiomedata/nmdc-schema/pull/457/files).

I expect to update the enumeration names soon, too, but I may be a little slower and more conservative with the permissible values of each enumeration. But those will also have to be dealt with eventually.

I would really appreciate it if you would at least look at the [Files Changed tab](https://github.com/microbiomedata/nmdc-schema/pull/457/files). Please let me know if you have a preferred time window for merging this into the main branch. I assume we will need one or two live discussions about this.

### CODE REVIEW

File: src/schema/annotation.yaml [hunk 1]
```diff
@@ -53,18 +53,18 @@ classes:
...
       start:
-        is_a: gff coordinate
+        is_a: gff_coordinate
         description: The start of the feature in positive 1-based integer coordinates
         required: true
         close_mappings:
           - biolink:start_interbase_coordinate
       end:
-        is_a: gff coordinate
+        is_a: gff_coordinate
```
Comment (cmungall) [inline src/schema/annotation.yaml:67]: Why not GffCoordinate?

Comment (cmungall) [reply src/schema/annotation.yaml:67]: Ignore my comment thought I was looking at classes

**Review by cmungall — APPROVED** [review]

File: src/schema/annotation.yaml (hunk 1 above)
Comment (scanon) [reply src/schema/annotation.yaml:67]: What did change in that line though?  Is it whitespace or something?

Comment (turbomam) [reply src/schema/annotation.yaml:67]: Thanks for looking @scanon. Please take a look at the [initial comment for this PR](https://github.com/microbiomedata/nmdc-schema/pull/457#issue-1370608167). Let me know if that doesn't answer your question.

**Comment by turbomam** [comment]
I am going to merge this PR and make nmdc-schema GH and PyPI releases after the "Metadata related" meeting today, unless anybody raises an objection before then.

**Review by sujaypatil96 — APPROVED** [review]

**Review by emileyfadrosh — APPROVED** [review]

**Pull request merged**